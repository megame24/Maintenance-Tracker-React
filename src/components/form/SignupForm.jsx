import React from 'react';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import Loading from '../Loading';
import ErrorMessage from '../ErrorMessage';

const SignupForm = (props) => {
  const {
    handleSubmit, formData, handleChange,
    isLoading, errors,
  } = props;
  return (
    <section id="register">
      <Loading isLoading={isLoading} />
      <div className="container">
        <ErrorMessage errors={errors} />
        <div className="form">
          <div className="form-icon-container">
            <div className="center form-icon blue">
              <i className="icon ion-person" />
            </div>
          </div>
          <h1 className="center">Create Account</h1>
          <form id="signupForm" onSubmit={handleSubmit}>
            <input
              onChange={handleChange}
              value={formData.name}
              className="form-input"
              name="fullname"
              type="text"
              id="name"
              placeholder="Full Name"
              required
            />
            <input
              onChange={handleChange}
              value={formData.email}
              className="form-input"
              type="text"
              name="email"
              id="email"
              placeholder="Email"
              required
            />
            <input
              onChange={handleChange}
              value={formData.username}
              className="form-input"
              type="text"
              name="username"
              id="username"
              placeholder="Username"
              required
            />
            <input
              onChange={handleChange}
              value={formData.password}
              className="form-input"
              type="password"
              name="password"
              id="password"
              placeholder="Password"
              required
            />
            <input
              className="form-input btn btn-primary"
              type="submit"
              id="submit-btn"
              value="Create Account"
            />
          </form>
          <div className="form-footer center">
            <p>
              Already a member? &nbsp;
              <Link to="/login" className="blue">Login</Link>
            </p>
          </div>
        </div>
      </div>
    </section>
  );
};

SignupForm.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  formData: PropTypes.shape({
    email: PropTypes.string.isRequired
  }).isRequired,
  isLoading: PropTypes.bool.isRequired,
  errors: PropTypes.shape({
    statusCode: PropTypes.number
  }),
};

SignupForm.defaultProps = {
  errors: {},
};

export default SignupForm;
