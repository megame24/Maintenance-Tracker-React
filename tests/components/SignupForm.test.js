import React from 'react';
import { shallow } from 'enzyme';
import SignupForm from '../../src/components/form/SignupForm';

describe('testing SignupForm component', () => {
  const props = {
    isLoading: false,
    handleChange: () => {},
    handleSubmit: () => {},
    formData: {
      fullname: '',
      password: '',
      username: '',
      email: '',
    }
  }
  it('should render as expected when there is no error', () => {
    const tree = shallow(<SignupForm {...props} />)
    expect(tree).toMatchSnapshot();
  });
  it('should render as expected when there is an error', () => {
    props.errors = {
      message: 'error'
    };
    const tree = shallow(<SignupForm {...props} />)
    expect(tree).toMatchSnapshot();
  });
});

