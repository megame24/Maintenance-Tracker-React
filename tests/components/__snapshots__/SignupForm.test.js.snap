// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SignupForm component should render as expected when there is an error 1`] = `
<section
  id="register"
>
  <Dimmer
    active={false}
    page={true}
  >
    <Loader
      size="medium"
    />
  </Dimmer>
  <div
    className="container"
  >
    <div
      className="message error-message"
    >
      error
    </div>
    <div
      className="form"
    >
      <div
        className="form-icon-container"
      >
        <div
          className="center form-icon blue"
        >
          <i
            className="icon ion-edit"
          />
        </div>
      </div>
      <h1
        className="center"
      >
        Create Account
      </h1>
      <form
        id="signupForm"
        onSubmit={[Function]}
      >
        <input
          className="form-input"
          id="name"
          name="fullname"
          onChange={[Function]}
          placeholder="Full Name"
          required={true}
          type="text"
        />
        <input
          className="form-input"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="text"
          value=""
        />
        <input
          className="form-input"
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
        <input
          className="form-input"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <input
          className="form-input btn btn-primary"
          id="submit-btn"
          type="submit"
          value="Create Account"
        />
      </form>
      <div
        className="form-footer center"
      >
        <p>
          Already a member?  
          <Link
            className="blue"
            replace={false}
            to="/login"
          >
            Login
          </Link>
        </p>
      </div>
    </div>
  </div>
</section>
`;

exports[`testing SignupForm component should render as expected when there is no error 1`] = `
<section
  id="register"
>
  <Dimmer
    active={false}
    page={true}
  >
    <Loader
      size="medium"
    />
  </Dimmer>
  <div
    className="container"
  >
    <div
      className="form"
    >
      <div
        className="form-icon-container"
      >
        <div
          className="center form-icon blue"
        >
          <i
            className="icon ion-edit"
          />
        </div>
      </div>
      <h1
        className="center"
      >
        Create Account
      </h1>
      <form
        id="signupForm"
        onSubmit={[Function]}
      >
        <input
          className="form-input"
          id="name"
          name="fullname"
          onChange={[Function]}
          placeholder="Full Name"
          required={true}
          type="text"
        />
        <input
          className="form-input"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="text"
          value=""
        />
        <input
          className="form-input"
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
        <input
          className="form-input"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <input
          className="form-input btn btn-primary"
          id="submit-btn"
          type="submit"
          value="Create Account"
        />
      </form>
      <div
        className="form-footer center"
      >
        <p>
          Already a member?  
          <Link
            className="blue"
            replace={false}
            to="/login"
          >
            Login
          </Link>
        </p>
      </div>
    </div>
  </div>
</section>
`;
